MediaGalleryCmsUi/view/adminhtml/web/js/form/element/used-in-ui-select.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'Magento_Ui/js/form/element/ui-select',
+    'jquery',
+    'underscore'
+], function (Select, $, _) {
+    'use strict';
+
+    return Select.extend({
+
+        /**
+         * @inheritdoc
+         */
+        success: function (response) {
+            var options = [];
```

Comment:
Options reset at each request success will most likely break the pagination functionality:
 - Generate more than 50 products (`bin/magento setup:perf:generate-fixtures setup/performance-toolkit/profiles/ce/small.xml`
 - Go to media gallery and type "Product" to "Used in Products" filter search field
 - Scroll to the bottom of the loaded options in the select
 - Next 50 products should be loaded

(it's easier to reproduce with products as we have the fixture to generate them, but the same behaviour should be preserved for all entities)